app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java

```diff
@@ -112,23 +112,75 @@ protected void onCreate(Bundle savedInstanceState) {
 
         addExitButton();
         addQuestionButton();
+        if(useDB){
+            addDBProxyGame();
+        }
     }
 
+    /**
+     * @param gameId The game ID to fetch the game from the DB
+     *    place the coins that have juste been created by placeCoins in the DB
+     *    adds a listener for the coins
+     */
     public void initializeGame(String gameId){
-        GameDatabaseProxy proxyG = new GameDatabaseProxy();
         proxyG.getGameDataSnapshot(gameId).addOnCompleteListener(task -> {
-        if (task.isSuccessful()) {
-            game = proxyG.getGameFromTaskSnapshot(task);
+            if (task.isSuccessful()) {
+                game = proxyG.getGameFromTaskSnapshot(task);
 
-            if(player.equals(game.getHost())){
-                game.setCoins(localPlayer.getLocallyAvailableCoins(), false);
-            }else{
+                if(player.equals(game.getHost())){
+                    game.setCoins(localPlayer.getLocallyAvailableCoins(), false);
+                    addCoinsListener();
+                }else{
+                }
+            } else {
+                Log.e(Game.class.getSimpleName(), task.getException().getMessage());
+            }
+        });
+    }
 
+    /**
+     * Adds a database proxy to the map
```

Comment:
I think the comment and method name are poorly phrased. If I'm not mistaken, it seems that this method get the game from the db.